lib/view_model/active_record.rb

```diff
@@ -165,27 +160,20 @@ def load(scope: nil, eager_include: true, serialize_context: new_serialize_conte
       vms
     end
 
-    def deserialize_from_view(subtree_hashes, references: {}, deserialize_context: new_deserialize_context)
+    def deserialize_from_view(subtree_hash_or_hashes, references: {}, deserialize_context: new_deserialize_context)
       model_class.transaction do
-        return_array = subtree_hashes.is_a?(Array)
-        subtree_hashes = Array.wrap(subtree_hashes)
-
-        root_update_data, referenced_update_data = UpdateData.parse_hashes(subtree_hashes, references)
-
-        # Provide information about will was updated
-        deserialize_context.updated_associations = root_update_data
-                                                     .map { |upd| upd.updated_associations }
-                                                     .inject({}) { |acc, assocs| acc.deep_merge(assocs) }
-
-        updated_viewmodels =
-          UpdateContext
-            .build!(root_update_data, referenced_update_data, root_type: self)
-            .run!(deserialize_context: deserialize_context)
-
-        if return_array
-          updated_viewmodels
-        else
-          updated_viewmodels.first
+        ViewModel::Utils.wrap_one_or_many(subtree_hash_or_hashes) do |subtree_hashes|
+          root_update_data, referenced_update_data = UpdateData.parse_hashes(subtree_hashes, references)
+
+          # Provide information about what was updated
+          deserialize_context.updated_associations = root_update_data
+                                                       .map { |upd| upd.updated_associations }
+                                                       .inject({}) { |acc, assocs| acc.deep_merge(assocs) }
+
+          _updated_viewmodels =
```

Comment:
Do you favour an effect-less assignment as documentation of return value significance?